pycodestyle.py

```diff
@@ -1,4 +1,5 @@
 #!/usr/bin/env python
+# -*- coding: utf8 -*-
```

Comment:
this shouldn't be needed now

Reply:
https://github.com/PyCQA/pycodestyle/pull/1035/commits/61a8e66d5289f54155a4bc14dd1f0a7a9661315c

Reply:
I reverted the commit as we need the encoding for examples in docstrings.

Reply:
the examples in the docstrings aren't readable, I'd prefer writing `\xa0` there as well

Reply:
Would you rather have no docstring exemples with NBSP or some hard to read one ? Because ``    E262: x = x + 1  # \xa0Increment x`` does not trigger E262 (a real one will).

Reply:
there's other examples where `\t` is used similarly, putting `E262: x = x + 1 # \xa0Increment x` (literally) into the docstring I believe gets the point across and is readable

Reply:
There's a self test that check that the example are working apparently:
```
____________________________________________________________________________________ PycodestyleTestCase.test_selftest ____________________________________________________________________________________

self = <testsuite.test_all.PycodestyleTestCase testMethod=test_selftest>

    def test_selftest(self):
        fail_s, done_s = selftest(self._style.options)
        self.assertTrue(done_s, msg='tests not found')
>       self.assertFalse(fail_s, msg='%s failure(s)' % fail_s)
E       AssertionError: 1 is not false : 1 failure(s)

testsuite/test_all.py:33: AssertionError
------------------------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------------------------
pycodestyle.py: failed to find E262:
x = x + 1  # \xa0Increment x
```

Reply:
if you remove the `r` on that docstring it should fix it

Reply:
Done in https://github.com/PyCQA/pycodestyle/pull/1035/commits/fe23aa2344d6f6b81e968b60fb4597d3c0d61347